### PR TITLE
Continuous Gradle Build and Maven Central Publishing

### PR DESCRIPTION
**Pull request by thehoneymad**
### TL; DR
These changes enable continuous gradle build on any commit to **any** branch through `build.yml` workflow. For `develop` branch, a snapshot release will be published through `release.yml`. For main, a release cut through tags will be release a release jar to maven central.

### More
This is the first PR to turn this repo follow basic gitflow. We are starting with a `main` branch and a `develop` branch. Feature branches will spawn off `develop` branch and merge into `develop`. When `develop` is stable enough, it will merge to main to start an official release.

### How this is built and released?
1. A generic `build.yml` will take care of building in PR requests and pushes.
2. A scoped down `release.yml` will take care of publishing snapshot release from `develop` branch and general release from `main` branch.

With time we will solidify the process with release notes, automatic changelog updates and a dedicated process.

I also added a contribution and developer guide here: https://github.com/maplibre/maplibre-java/wiki

### CODE REVIEW

File: .github/workflows/test-release.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        run: |
+          ./gradlew build --info --stacktrace
+      - name: Validate Gradle wrapper
+        uses: gradle/wrapper-validation-action@e6e38bacfdf1a337459f332974bb2327a31aaf4b
+      - name: Publish package
+        uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
+        with:
+          arguments: publish
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (wipfli) [inline .github/workflows/test-release.yml:31]: Looks like it actually publishes the package. From the name of the workflow I thought this would be `test-release.yml` meaning "test if the release would work"...

File: .github/workflows/develop.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Development build - Linux
```
Comment (wipfli) [inline .github/workflows/develop.yml:1]: I would call this workflow `build.yml`

Comment (thehoneymad) [reply .github/workflows/develop.yml:1]: Ah, okay. Can do!

File: .github/workflows/test-release.yml (hunk 1 above)
Comment (thehoneymad) [reply .github/workflows/test-release.yml]: Yeah. This is definitely a test release. :) But it publishes to Github Packages. Not the Maven Central. If this works out, I will add Maven Central to it.

**Review by artakka — APPROVED** [review]
We will need to document this somewhere, correct?

**Comment by thehoneymad** [comment]
> 

Yes. I am hoping to do that in the wiki. :)

**Comment by thehoneymad** [comment]
I cut issue #13 to address the formal release problem

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  pull_request:
+    types: [opened, reopened, edited]
+  push:
+    paths-ignore:
+      - "README.md"
+      - ".gitignore"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (wipfli) [inline .github/workflows/build.yml:16]: probably best to pin the ubuntu version to for example `ubuntu-20.04` and then use dependabot to update the workflows, see https://github.com/maplibre/maplibre-gl-js/blob/11a1262601d9842efb4f6a44fed40fcb55fc2462/.github/dependabot.yml#L14 for an example

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -1,56 +1,56 @@
...
+  IS_SNAPSHOT: ${{ github.ref_name != 'main' }}
 on:
   workflow_dispatch:
   push:
     tags:
-      - v* 
-    paths-ignore:      
-      - ".github/**"      
-      - "README.md"
+      - v*
```
Comment (wipfli) [inline .github/workflows/release.yml:9]: Triggering on tag push was somehow always a pain in MapLibre GL JS, because when stuff fails, the tags are still there and things get messy somehow.

I would recommend to trigger the release workflow only on `workflow_dispatch` and then the action can tag the git commit with a version. You can have a look how we do this in MapLibre GL JS at

https://github.com/maplibre/maplibre-gl-js/blob/11a1262601d9842efb4f6a44fed40fcb55fc2462/.github/workflows/release.yml#L34-L38

**Comment by wipfli** [comment]
Looks good in general, more automation is better I think. If we have the opportunity to automate things in the `maplibre-java` repo I would suggest to either follow the same path as MapLibre GL JS, where things are quite automated but changelogs are still manual, or that we just copy completely what @smelleyshovel does for MapLibre GL Directions.

There you have

* automated versioning based on GitHub tags of pull requests (if you tag a pull request with `semver:minor`, it will bumpt the minor version after merging automatically...)
* automated changelog based on pull request titles I think

**Comment by wipfli** [comment]
https://github.com/maplibre/maplibre-gl-directions/tree/main/.github/workflows
